flink-runtime/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java

```diff
@@ -1018,4 +1053,21 @@ public QueryableStateStream<KEY, T> asQueryableState(
                 stateDescriptor,
                 getKeyType().createSerializer(getExecutionConfig().getSerializerConfig()));
     }
+
+    /**
+     * Enable the async state processing for following keyed processing function. This also requires
+     * only State V2 APIs are used in the function.
+     *
+     * @return the configured KeyedStream itself.
+     */
+    @Experimental
```

Comment:
Does the queryable state need to support async state?

Reply:
Well, since the queryable state is deprecated, we won't support that.